### PR TITLE
Rework long.ToMetric for correct interpretation of decimals parameter

### PR DESCRIPTION
**Pull request by logiclrd**
This PR:

- Replaces the test suite for the long.ToMetric extension method with a much more comprehensive set using a `TheoryData` generator, and uses an interpretation of the `decimals` parameter in line with `int.ToMetric` and `double.ToMetric` in the process.

- Makes all the tests pass by reworking the implementation of the `ToMetric(long, ..)` extension method and its underlying helper `BuildRepresentation(long, ..)` correspondingly.

This amends the implementation merged in #1596 to correct this misinterpretation. Thanks to @petervanleeuwen for noticing it 🙂 

Here is a checklist you should tick through before submitting a pull request: 
 - [x] Implementation is clean
 - [x] Code adheres to the existing coding standards; e.g. no curlies for one-line blocks, no redundant empty lines between methods or code blocks, spaces rather than tabs, etc.
     - In `ToMetric(this long, ..)`, there is a one-line block with braces. The reason is consistency with the existing code in `ToMetric(this int, ..)` and `ToMetric(this double, ..)`. If you prefer, I can add a commit that updates all of them to follow this style point.
 - [x] No Code Analysis warnings
 - [x] There is proper unit test coverage
 - [x] (N/A) If the code is copied from StackOverflow (or a blog or OSS) full disclosure is included. That includes required license files and/or file headers explaining where the code came from with proper attribution
 - [x] There are very few or no comments (because comments shouldn't be needed if you write clean code)
    - I put a comment next to a loop whose purpose isn't hard to discern but for which understanding how one iteration's effects tie into the next would add a handful of unnecessary seconds to the reader's day.
    - I used comments to denote number ranges in the test case generator. The separation reads better to my eye. Can remove them if they're deemed inappropriate.
 - [x] (N/A) Xml documentation is added/updated for the addition/change
 - [x] Your PR is (re)based on top of the latest commits from the `main` branch (more info below)
 - [x] Link to the issue(s) you're fixing from your PR description. Use `fixes #<the issue number>`
 - [x] (N/A) Readme is updated if you change an existing feature or add a new one
 - [x] Run either `build.cmd` or `build.ps1` and ensure there are no test failures
    - I don't see `build.cmd` or `build.ps1` in the source tree, but the solution builds and all tests discovered by the Test Explorer pass.

Fixes: #1699

### CODE REVIEW

**Comment by logiclrd** [comment]
Rebased.

**Comment by logiclrd** [comment]
@clairernovotny Is there any chance this PR might be considered? 🙂

File: src/Humanizer/MetricNumeralExtensions.cs [hunk 1]
```diff
@@ -337,41 +328,62 @@ static string BuildMetricRepresentation(long input, int scale, MetricNumeralForm
...
+            }
+
+            while ((decimals > 0) && (fractionalPartCharacters[decimals.Value - 1] == '0'))
+                decimals--;
+        }
 
         if (decimals == 0)
         {
+            if ((divisor > 1) && (fractionalPart >= divisor / 2))
+                number += Math.Sign(number);
```
[the remark this answers is not in the document]
Comment (clairernovotny) [reply src/Humanizer/MetricNumeralExtensions.cs]: Already addressed on the current head: the `decimals == 0` path applies ToEven only when `fractionalPart == roundingPoint`, using the parity of the retained integer. The generated test data also includes explicit midpoint rows for positive and negative values.

File: src/Humanizer/MetricNumeralExtensions.cs [hunk 2]
```diff
@@ -337,41 +328,72 @@ static string BuildMetricRepresentation(long input, int scale, MetricNumeralForm
...
+
+            if (divisor > 1)
+            {
+                if (fractionalPart > roundingPoint)
+                    number += Math.Sign(number);
+                else if (fractionalPart == roundingPoint)
+                {
+                    // Use banker's rounding for consistency with Math.Round used elsewhere on floats.
+                    number += Math.Sign(number % 2);
+                }
```
[the remark this answers is not in the document]
Comment (clairernovotny) [reply src/Humanizer/MetricNumeralExtensions.cs:446]: Already addressed on the current head: after zero-decimal rounding, `Math.Abs(number) == 1000` promotes the value and selects the next metric unit before formatting. The generated tests include positive and negative scale-boundary cases such as `999500 → 1M`.

**Comment by clairernovotny** [comment]
Addressed the remaining review items:

- 64a52c8f updates the public `long.ToMetric` XML docs to describe `decimals` as maximum fractional precision, the `null` behavior, and trailing-zero trimming.
- The generated expectations remain independent of the production digit/carry implementation: they use decimal scaling plus `Math.Round`, with explicit midpoint and scale-transition cases, so I kept that oracle rather than duplicating the production algorithm.

Validation passed on .NET 8, 10, and 11 (56,970 tests each), plus Release packing and full format verification.

File: tests/Humanizer.Tests/MetricNumeralTests.cs [hunk 3]
```diff
@@ -75,6 +75,174 @@ public void TestAllSymbolsAsInt(int exponent)
...
+                var scaledValue = value * 0.001M;
+
+                var truncatedValue =
+                    decimals.HasValue
+                    ? Math.Round(scaledValue, decimals.Value)
+                    : scaledValue;
+
+                RemoveTrailingZeroes(ref truncatedValue);
+
+                var expected = truncatedValue + "k";
```
[the remark this answers is not in the document]
Comment (clairernovotny) [reply tests/Humanizer.Tests/MetricNumeralTests.cs]: Addressed in 25962432: every generated expected number now formats with `CultureInfo.InvariantCulture`, including unscaled values and all metric suffixes. A `de-DE` regression directly invokes the data generator and verifies `1245` at two decimals produces `1.24k`.

**Comment by clairernovotny** [comment]
@coderabbitai review
